### ISSUE TITLE
[SCIM] Errors response body not according to the RFC

### ISSUE THREAD

**Issue by ZahiC** [simpleidserver/SimpleIdServer#227]
Hi, I tried to run [OKTA Runscope tests](https://developer.okta.com/docs/guides/scim-provisioning-integration-prepare/main/#test-your-scim-api) against Simple ID SCIM server. Most of the tests pass.

It failed on 404 error. For getting non-existing resource (user), it expects:
```json
{
  "status": "404",
  "schemas": ["urn:ietf:params:scim:api:messages:2.0:Error"],
  "detail": "resource X not found"
}
```

While Simple ID response is:
```json
{
  "status": "404",
  "response": {
    "schemas": ["urn:ietf:params:scim:api:messages:2.0:Error"],
    "status": "404",
    "detail": "resource X not found"
  }
}
```

It seems that the RFC expect the error response to be on the root of the response json: 

![image](https://user-images.githubusercontent.com/15949736/157001650-8dca2ca2-3b36-44b7-9388-ac030ced29c7.png)

I think it's also relevant for other error responses in the SCIM server.

**Comment by simpleidserver** [comment]
Hello,

Indeed there is a small issue in our SCIM implementation :)
Thank you very much for your feedback ! We are going to fix it ASAP.

Kind Regards,

SimpleIdServer

**Comment by simpleidserver** [comment]
The issue is fixed in the branch "release/2.0.5".

Kind Regards,

SimpleIdServer
